src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1210,6 +1210,14 @@ void VM_Version::print_using_features_cr() {
   }
 }
 
+bool VM_Version::nonlibc_str_equals(const char *a, const char *b) {
+  while (*a && *b) {
+    if (*a++ != *b++)
+      return false;
+  }
+  return !*a && !*b;
+}
```

Comment:
Compiler can inline a call to strcmp if it recognizes the pattern, so this is not really safe. And since we are calling the argument check after VM_Crac::read_shm, it's a bit late to worry about libc... It may be just fine to leave the one who are give up the safety with possible crashes.

Reply:
That was a gross mistake in this patch, I have fixed it so that it no longer crashes. That was the whole effort of the original patch so that it does not crash without any error if it is being restored on under-featured CPU.